core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -189,13 +190,12 @@ private[spark] class HighlyCompressedMapStatus private (
     emptyBlocks.readExternal(in)
     avgSize = in.readLong()
     val count = in.readInt()
-    val hugeBlockSizesArray = mutable.ArrayBuffer[Tuple2[Int, Byte]]()
+    hugeBlockSizes = new util.HashMap[Int, Byte](count).asScala
```

Comment:
we should use a `mutable.Map` instead of the java implementation wrapped by scala, this is not a clean solution IMHO